lib/protocol/SqlString.js

```diff
@@ -72,20 +73,41 @@ SqlString.arrayToList = function arrayToList(array, timeZone) {
 };
 
 SqlString.format = function format(sql, values, stringifyObjects, timeZone) {
-  values = values == null ? [] : [].concat(values);
+  if (values == null) {
+    return sql;
+  }
 
-  var index = 0;
-  return sql.replace(/\?\??/g, function(match) {
-    if (index === values.length) {
-      return match;
-    }
+  if (!(values instanceof Array)) {
```

Comment:
> Well I'd hope unit tests would enforce that the behavior stays the same

If I had written the module from the ground up, it probably would :) But we have what we have until it's improved upon :)

So, specifically, what I'm asking is that now we are not doing `values = [].concat(values)` and instead doing `values = values instanceof Array ? values = [values]`. Does the `instanceof` check exactly match what `.concat()` is doing interally? You brought up the different from the `vm` (arrays from a different context). We we get one of those arrays, where `instanceof Array` is `false`, do we still behave the same as the `.concat` pattern did?